Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -0,0 +1,753 @@
+        shape.transform = line.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: line.position, initialOpacity: line.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addDashedCircle(circle: WelcomeCircle, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_circlePathWithRadius(toView.frame.size.width * circle.radius).CGPath
+        shape.fillColor = UIColor.clearColor().CGColor
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.lineDashPattern = [toView.frame.size.width * 0.029, toView.frame.size.width * 0.047]
+        shape.transform = circle.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position, initialOpacity: circle.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addCircle(circle: WelcomeCircle, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_circlePathWithRadius(toView.frame.size.width * circle.radius).CGPath
+        shape.fillColor = UIColor.blackColor().CGColor
+        shape.strokeColor = UIColor.clearColor().CGColor
+        shape.transform = circle.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position, initialOpacity: circle.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addPlus(plus: WelcomePlus, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_plusPathWithWidth(toView.frame.size.width * 0.04).CGPath
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.transform = plus.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: plus.position, initialOpacity: plus.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addBar(bar: WelcomeBar, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        let rect = CGRectMake(
+            0.0,
+            0.0,
+            toView.frame.size.width * bar.frame.size.width,
+            toView.frame.size.height * bar.frame.size.height
+        )
+        shape.path = UIBezierPath(roundedRect: rect, cornerRadius: bar.cornerRadius).CGPath
+        shape.fillColor = bar.color.CGColor
+        shape.transform = bar.animation.initialTransform
+        shape.position = CGPointMake(
+            (toView.frame.size.width * bar.frame.origin.x),
+            (toView.frame.size.height * bar.frame.origin.y)
+        )
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+}
+
+extension CALayer{
+    func wmf_addWelcomeAnimation(animation: WelcomeAnimation) {
```

Comment:
not very descriptive. Try to use names that describes what is happening - as opposed to `welcome…`